pkg/workspace/plugins.go

```diff
@@ -409,6 +424,80 @@ func GetPluginPath(kind PluginKind, name string, version *semver.Version) (strin
 	return "", "", nil
 }
 
+type sortedPluginInfo []PluginInfo
+
+func (sp sortedPluginInfo) Len() int { return len(sp) }
+func (sp sortedPluginInfo) Less(i, j int) bool {
+	iVersion := sp[i].Version
+	jVersion := sp[j].Version
+	switch {
+	case iVersion == nil && jVersion == nil:
+		return false
+	case iVersion == nil:
+		return true
+	case jVersion == nil:
+		return false
+	default:
+		return iVersion.LT(*jVersion)
+	}
+}
+func (sp sortedPluginInfo) Swap(i, j int) { sp[i], sp[j] = sp[j], sp[i] }
+
+// SelectCompatiblePlugin selects a plugin from the list of plugins with the given kind and name that sastisfies the
+// requested semver range. It returns the highest version plugin that satisfies the requested constraints, or an error
+// if no such plugin could be found.
+//
+// If there exist plugins in the plugin list that don't have a version, SelectCompatiblePlugin will select them if there
+// are no other compatible plugins available.
```

Comment:
For my own edification: When can a plugin not have a version?

Reply:
From an in person discussion the general answer seems to be: "when they were built the wrong way".